src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,195 @@
+import abc
+import logging
+import os
+from typing import Dict, Union, Optional, Tuple
+from pathlib import Path
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+from funcy import merge_with
+
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 20
+
+METRIC_CPU_COUNT = "system/cpu/count"
+METRIC_CPU_USAGE_PERCENT = "system/cpu/usage (%)"
+METRIC_CPU_PARALLELIZATION_PERCENT = "system/cpu/parallelization (%)"
+
+METRIC_RAM_USAGE_PERCENT = "system/ram/usage (%)"
+METRIC_RAM_USAGE_GB = "system/ram/usage (GB)"
+METRIC_RAM_TOTAL_GB = "system/ram/total (GB)"
+
+METRIC_DISK_USAGE_PERCENT = "system/disk/usage (%)"
+METRIC_DISK_USAGE_GB = "system/disk/usage (GB)"
+METRIC_DISK_TOTAL_GB = "system/disk/total (GB)"
+
+
+class _SystemMonitor(abc.ABC):
+    """
+    Monitor system resources and log them to DVC Live.
+    Use a separate thread to call a `_get_metrics` function at fix interval and
+    aggregate the results of this sampling using the average.
+    """
+
+    _plot_blacklist_prefix: Tuple = ()
+
+    def __init__(
+        self,
+        interval: float = 0.5,
+        num_samples: int = 10,
+        plot: bool = True,
+    ):
+        if not isinstance(interval, (int, float)):
```

Comment:
hmm, I don't think we are input types like this

Reply:
in this case the idea was to test probably that value are meaningful and raise ValueError? or ignore CPU monitoring

Reply:
Oh ok I see. 
I changed to keep the monitoring even if the values are out of range, but I added a warning message. 
I took the min and max values from here: https://github.com/wandb/wandb/blob/852781a852a5ae63ea009b40fe923e1a80603fd0/wandb/sdk/internal/system/assets/interfaces.py#L120